src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -52,17 +64,42 @@ public DrsService(SnapshotService snapshotService,
                       DrsIdService drsIdService,
                       IamService samService,
                       ResourceService resourceService,
+                      ConfigurationService configurationService,
+                      KubeService kubeService,
                       PerformanceLogger performanceLogger) {
         this.snapshotService = snapshotService;
         this.fileService = fileService;
         this.drsIdService = drsIdService;
         this.samService = samService;
         this.resourceService = resourceService;
+        this.configurationService = configurationService;
+        this.kubeService = kubeService;
         this.performanceLogger = performanceLogger;
     }
 
-    public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String drsObjectId, Boolean expand) {
+    public void decrement() {
+        int newValue = currentDRSRequests.decrementAndGet(); // how do I make sure this never becomes negative?
+        currentDRSRequests.set(newValue);
+    }
+
+    public void increment() {
+        int newValue = currentDRSRequests.incrementAndGet();
+        currentDRSRequests.set(newValue);
+        checkLookupMax();
+    }
+
+    public void checkLookupMax() throws TooManyRequestsException {
+        // make sure not too many requests are being made at once
+        int podCount = kubeService.getActivePodCount();
+        int maxDRSLookups = configurationService.getParameterValue(ConfigEnum.DRS_LOOKUP_MAX);
+        int max = maxDRSLookups / podCount;
+        if (currentDRSRequests.get() >= max) {
+            throw new TooManyRequestsException("Too many requests are being made at once. Please try again later.");
+        }
+    }
 
+    public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String drsObjectId, Boolean expand
+    ) throws TooManyRequestsException {
```

Comment:
Since this is unchecked you don't have a to declare it. It doesn't hurt to do it; conventionally it's more typical to javadoc unchecked exceptions rather than declare them in `throws`.